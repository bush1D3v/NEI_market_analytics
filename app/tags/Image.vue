<script lang="ts" setup>
import { computed, onMounted, ref, type HTMLAttributes, type ImgHTMLAttributes } from "vue";
import { cn } from "@/lib/utils";
import Skeleton from "@/components/ui/skeleton/Skeleton.vue";
import ImageSkeletonSvg from "@/components/Svg/ImageSkeletonSvg.vue";

export interface ImageProps extends /* @vue-ignore */ ImgHTMLAttributes {
    class?: HTMLAttributes[ "class" ];
    src: string;
    alt: string;
    width: string;
    height: string;
}

const props = defineProps<ImageProps>();

const loading = ref<boolean>(true);
const visible = ref<boolean>(true);

function onImageLoad() {
    loading.value = false;
}

const roundedClass = computed(() => {
    const classList = props.class?.split(" ") || [];
    return classList.filter((cls: string) => cls.startsWith("rounded")).join(" ");
});

onMounted(() => {
    setTimeout(() => {
        if (loading.value) {
            visible.value = false;
        }
    }, 5000);
});
</script>

<template>
    <div v-if="visible" class="relative" :class="`w-[${props.width}px] h-[${props.height}px] flex items-center`">
        <Skeleton v-if="loading" :class="`w-full h-full absolute ${roundedClass}`">
            <ImageSkeletonSvg />
        </Skeleton>
        <img :src="props.src" :alt="props.alt" :width="props.width" :height="props.height" :class="cn(props.class)"
            data-testid="Image" @load="onImageLoad()" />
    </div>
    <div v-else :class="` flex items-center justify-center`">
        <ImageSkeletonSvg :class="`w-[${props.width}px] h-[${props.height}px]`" />
    </div>
</template>
