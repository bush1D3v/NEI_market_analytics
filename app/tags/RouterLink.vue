<script lang="ts" setup>
import type {HTMLAttributes} from "vue";
import {cn} from "@/lib/utils";

export interface RouterLinkProps {
	class?: HTMLAttributes["class"];
	to: string;
}

const props = defineProps<RouterLinkProps>();
</script>

<template>
    <router-link :to="props.to" :class="cn('duration-150 ease-in-out cursor-pointer', props.class)"
        data-testid="RouterLink">
        <slot />
    </router-link>
</template>
