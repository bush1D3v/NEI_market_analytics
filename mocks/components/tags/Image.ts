import type {ImageProps} from "@/tags/Image.vue";

export const ImageMock: ImageProps = {
	alt: "image",
	height: "80",
	width: "80",
	src: "https://via.placeholder.com/80",
	class: "w-20 h-20 rounded-full",
};
