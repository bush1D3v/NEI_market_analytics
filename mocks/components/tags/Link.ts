import type {LinkProps} from "@/tags/Link.vue";

export const LinkMock: LinkProps = {
	href: "https://www.google.com",
	target: "_blank",
	class: "font-bold underline",
};
