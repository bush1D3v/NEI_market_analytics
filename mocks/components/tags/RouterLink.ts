import type {RouterLinkProps} from "@/tags/RouterLink.vue";

export const RouterLinkMock: RouterLinkProps = {
	to: "/about-us",
	class: "font-bold underline",
};
