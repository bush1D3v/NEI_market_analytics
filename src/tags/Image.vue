<script lang="ts" setup>
import { computed, ref, type HTMLAttributes, type ImgHTMLAttributes } from "vue";
import { cn } from "@/lib/utils";
import Skeleton from "@/components/ui/skeleton/Skeleton.vue";
import ImageSkeletonSvg from "@/components/Svg/ImageSkeletonSvg.vue";

export interface ImageProps extends /* @vue-ignore */ ImgHTMLAttributes {
    class?: HTMLAttributes[ "class" ];
    src: string;
    alt: string;
    width: string;
    height: string;
}

const props = defineProps<ImageProps>();

// Data
const loading = ref<boolean>(true);

// Methods
function onImageLoad() {
    loading.value = false;
}

const roundedClass = computed(() => {
    const classList = props.class?.split(" ") || [];
    return classList.filter((cls: string) => cls.startsWith("rounded")).join(" ");
});
</script>

<template>
    <div class="relative" :class="`w-[${props.width}px] h-[${props.height}px]`">
        <Skeleton v-if="loading" :class="`w-full h-full absolute ${roundedClass}`">
            <ImageSkeletonSvg />
        </Skeleton>
        <img :src="props.src" :alt="props.alt" :width="props.width" :height="props.height" :class="cn(props.class)"
            data-testid="Image" @load="onImageLoad()" />
    </div>
</template>
