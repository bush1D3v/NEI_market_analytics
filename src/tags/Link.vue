<script lang="ts" setup>
import type {AnchorHTMLAttributes, HTMLAttributes} from "vue";
import {cn} from "@/lib/utils";

export interface LinkProps extends /* @vue-ignore */ AnchorHTMLAttributes {
	class?: HTMLAttributes["class"];
	target: "_blank" | "_self" | "_parent" | "_top";
	href: string;
}

const props = defineProps<LinkProps>();
</script>

<template>
    <a :href="props.href" :target="props.target" rel="no-referrer" :class="cn(props.class)" data-testid="Link">
        <slot />
    </a>
</template>
