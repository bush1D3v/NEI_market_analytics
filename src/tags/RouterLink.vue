<script lang="ts" setup>
import type { HTMLAttributes } from "vue";
import { cn } from "@/lib/utils";

export interface LinkProps {
    class?: HTMLAttributes[ "class" ];
    to: string;
}

const props = defineProps<LinkProps>();
</script>

<template>
    <router-link :to="props.to" :class="cn('duration-150 ease-in-out cursor-pointer', props.class)"
        data-testid="RouterLink">
        <slot />
    </router-link>
</template>
