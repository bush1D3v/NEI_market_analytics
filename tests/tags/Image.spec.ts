import {test, expect, describe} from "vitest";
import Image from "@/tags/Image.vue";
import {ImageMock} from "../../mocks/components/tags/Image";
import {shallowMount} from "@vue/test-utils";

describe("Image Component Tests", () => {
	const wrapper = shallowMount(Image, {props: {...ImageMock}});
	const imageComponent = wrapper.findComponent(Image);

	test("Should be able to render the component correctly", async () => {
		expect(imageComponent).toBeDefined();
	});

	test("Should be able to snapshot the component correctly", async () => {
		expect(imageComponent.html()).toMatchSnapshot();
	});

	test("Should be able to render the component with the correct props", async () => {
		expect(imageComponent.props("src")).toBe(ImageMock.src);
		expect(imageComponent.props("alt")).toBe(ImageMock.alt);
		expect(imageComponent.props("width")).toBe(ImageMock.width);
		expect(imageComponent.props("height")).toBe(ImageMock.height);
		expect(imageComponent.props("class")).toBe(ImageMock.class);
	});
});
