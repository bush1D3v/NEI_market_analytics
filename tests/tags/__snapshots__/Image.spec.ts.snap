// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Image Component Tests > Should be able to snapshot the component correctly 1`] = `
"<div class="relative w-[80px] h-[80px]">
  <div class="animate-pulse bg-surface flex justify-center items-center w-full h-full absolute rounded-full"><svg class="w-8 h-8" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" fill="currentColor" viewBox="0 0 20 18" role="img" focusable="false" aria-labelledby="svgTitle svgDesc">
      <title id="svgTitle">Image Placeholder</title>
      <desc id="svgDesc">A placeholder image in the shape of a rectangle with a mountain icon.</desc>
      <path d="M18 0H2a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2Zm-5.5 4a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Zm4.376 10.481A1 1 0 0 1 16 15H4a1 1 0 0 1-.895-1.447l3.5-7A1 1 0 0 1 7.468 6a.965.965 0 0 1 .9.5l2.775 4.757 1.546-1.887a1 1 0 0 1 1.618.1l2.541 4a1 1 0 0 1 .028 1.011Z"></path>
    </svg></div><img src="https://via.placeholder.com/80" alt="image" width="80" height="80" class="w-20 h-20 rounded-full" data-testid="Image">
</div>"
`;
